src/braket/circuits/circuit.py

```diff
@@ -412,10 +426,28 @@ def add_instruction(
             # non single qubit operator with target, add instruction with target
             instructions_to_add = [instruction.copy(target=target)]
 
+        if self._check_for_params(instruction):
+            self._parameters.add(instruction.operator.parameter)
         self._moments.add(instructions_to_add)
 
         return self
 
+    def _check_for_params(self, instruction: Instruction) -> bool:
+        """
+        This checks for free parameters in an :class:{Instruction}. Checks children classes of
+        :class:{Parameterizable}.
+
+        Args:
+            instruction: The instruction to check for a :class:{FreeParameter}.
```

Comment:
Annotate with type; applies everywhere else too:
```suggestion
            instruction (Instruction): The instruction to check for a :class:{FreeParameter}.
```